dev/breeze/src/airflow_breeze/commands/ci_commands.py

```diff
@@ -423,3 +423,246 @@ def get_workflow_info(github_context: str, github_context_input: StringIO):
+        )
+        sys.exit(1)
+
+    # Check if we're on the main branch
+    branch_result = run_command(
+        ["git", "rev-parse", "--abbrev-ref", "HEAD"], capture_output=True, text=True, check=False
+    )
+    current_branch = branch_result.stdout.strip() if branch_result.returncode == 0 else ""
+
+    # Store the original branch/commit to restore later if needed
+    original_branch = current_branch
+    original_commit_result = run_command(
+        ["git", "rev-parse", "HEAD"], capture_output=True, text=True, check=False
+    )
+    original_commit = original_commit_result.stdout.strip() if original_commit_result.returncode == 0 else ""
+
+    # Check if the working directory is clean
+    status_result = run_command(["git", "status", "--porcelain"], capture_output=True, text=True, check=False)
+    is_clean = status_result.returncode == 0 and not status_result.stdout.strip()
+
+    # Check if we have the apache remote
+    remote_result = run_command(["git", "remote", "-v"], capture_output=True, text=True, check=False)
+    has_apache_remote = "apache/airflow" in remote_result.stdout if remote_result.returncode == 0 else False
+
+    # Check if we're up to date with apache/airflow on the specified branch
+    if has_apache_remote:
+        # Fetch apache remote to get latest info
+        run_command(["git", "fetch", "apache"], check=False)
+
+        # Check if current HEAD matches apache/<branch>
+        local_head = run_command(["git", "rev-parse", "HEAD"], capture_output=True, text=True, check=False)
+        remote_head = run_command(
+            ["git", "rev-parse", f"apache/{target_branch}"], capture_output=True, text=True, check=False
+        )
+
+        at_apache_branch = (
+            current_branch == target_branch
+            and local_head.returncode == 0
+            and remote_head.returncode == 0
+            and local_head.stdout.strip() == remote_head.stdout.strip()
+        )
+    else:
+        at_apache_branch = False
+        get_console().print("[warning]No apache remote found. Expected remote pointing to apache/airflow.[/]")
+
+    # Track whether user chose to reset to target branch
+    user_switched_to_target = False
+
+    if not at_apache_branch or not is_clean:
+        get_console().print()
+        if not at_apache_branch:
+            get_console().print(
+                f"[warning]You are not at the top of apache/airflow {target_branch} branch.[/]"
+            )
+            get_console().print(f"[info]Current branch: {current_branch}[/]")
+        if not is_clean:
+            get_console().print("[warning]Your repository has uncommitted changes.[/]")
+        get_console().print()
+
+        if (
+            response := user_confirm(
+                f"Do you want to reset to latest apache/airflow {target_branch} and clean up everything? "
+                "This will discard all local changes! No will continue anyway, quit will exit."
+            )
+            == Answer.YES
```

Comment:
```suggestion
            response := user_confirm(
                f"Do you want to reset to latest apache/airflow {target_branch} and lose non committed changes? "
                "This will discard all local changes! No will continue anyway, quit will exit."
            )
            == Answer.YES
```